### ISSUE TITLE
Fill constructor shortcut with class name and move cursor to parameters

### ISSUE THREAD

**Issue by CarlosTorrecillas** [OmniSharp/omnisharp-vscode#643]
## Environment data
`dotnet --info` output:
.NET Command Line Tools (1.0.0-preview2-003121)

Product Information:
 Version:            1.0.0-preview2-003121
 Commit SHA-1 hash:  1e9d529bc5

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  10.11
 OS Platform: Darwin
 RID:         osx.10.11-x64
VS Code version: 1.4.1
C# Extension version: 1.3.0

## Steps to reproduce

- Create a new class
- type the shortcut for constructor: ctor

## Expected  behavior

- I should have the class name in the constructor and should not have to tab to give the class name then tab again to go to the Parameters section which IMO should be blank

## Actual behavior

- I have the following code generated:
`public ClassName(Parameters)`

**Comment by DustinCampbell** [comment]
I tend to agree. The `ctor` snippet that was added is quite a bit different from the one that already exists in Visual Studio, which looks like this:

```
public $classname$ ()
{
    $end$
}
```

@jchannon: How do you feel about this (since you added the `ctor` snippet with #549)? Are you OK if we change it to better match the snippet in Visual Studio?

**Comment by jchannon** [comment]
Yup make sense. Go nuts

**Comment by CarlosTorrecillas** [comment]
Awesome, thanks!

**Comment by DustinCampbell** [comment]
FWIW, getting this working is tricky because VS Code appears to only support inserting text with placeholders as part of snippet expansion. To retrieve the class name, we'll need to be able to run code as the snippet expands. I've filed https://github.com/Microsoft/vscode/issues/10561 to track that.
